pkg/bigquery/utils/utils.go

```diff
@@ -93,3 +93,69 @@ func SendResponse(res interface{}, err error, rw http.ResponseWriter) {
 	rw.Header().Add("Content-Type", "application/json")
 	WriteResponse(rw, bytes)
 }
+
+// ErrorResponse represents a structured error response
+type ErrorResponse struct {
+	Error   string `json:"error"`
+	Message string `json:"message,omitempty"`
+	Code    int    `json:"code,omitempty"`
+}
+
+// HandleError processes an error, logs it appropriately, and returns a structured error response
+// It specifically handles Google API errors with enhanced logging and proper HTTP status codes
+func HandleError(err error, context string) (ErrorResponse, int) {
+	if err == nil {
+		return ErrorResponse{}, 0
+	}
+
+	// Check if it's a Google API error
+	if googleApiError, ok := err.(*googleapi.Error); ok {
+		// Log Google API error with details
+		log.DefaultLogger.Error("Google API error",
+			"context", context,
+			"code", googleApiError.Code,
+			"message", googleApiError.Message,
+			"details", googleApiError.Details,
+		)
+
+		// Return structured response with Google API error details
+		return ErrorResponse{
+			Error:   googleApiError.Message,
+			Message: fmt.Sprintf("Google API error: %s", googleApiError.Message),
+			Code:    googleApiError.Code,
+		}, googleApiError.Code
```

Comment:
Not sure if we need it here, but we could add also error source handling. If so, we should do it based on the code using something like `backend.ErrorSourceFromHTTPStatus(res.StatusCode)`